impexp-client/src/main/java/org/citydb/cli/DeleteCommand.java

```diff
@@ -0,0 +1,122 @@
+import org.citydb.plugin.cli.DatabaseOption;
+import org.citydb.registry.ObjectRegistry;
+import picocli.CommandLine;
+
+@CommandLine.Command(
+        name = "delete",
+        description = "Deletes top-level city objects from the database.",
+        versionProvider = ImpExpCli.class
+)
+public class DeleteCommand extends CliCommand {
+    enum Mode {delete, terminate};
+
+    @CommandLine.Option(names = {"-m", "--delete-mode"}, paramLabel = "<mode>", defaultValue = "delete",
+            description = "Delete mode: ${COMPLETION-CANDIDATES} (default: ${DEFAULT-VALUE}).")
+    private Mode mode;
+
+    @CommandLine.ArgGroup(exclusive = false, heading = "Query and filter options:%n")
+    private QueryOption queryOption;
+
+    @CommandLine.ArgGroup(exclusive = false, heading = "Delete list options:%n")
+    private DeleteListOption deleteListOption;
+
+    @CommandLine.ArgGroup(exclusive = false, heading = "Database connection options:%n")
+    private DatabaseOption databaseOption;
+
+    private final Logger log = Logger.getInstance();
+
+    @Override
+    public Integer call() throws Exception {
+        Config config = ObjectRegistry.getInstance().getConfig();
+
+        // connect to database
+        DatabaseController database = ObjectRegistry.getInstance().getDatabaseController();
+        DatabaseConnection connection = databaseOption != null ?
+                databaseOption.toDatabaseConnection() :
+                config.getDatabaseConfig().getActiveConnection();
+
+        if (!database.connect(connection)) {
+            log.warn("Database delete aborted.");
+            return 1;
+        }
+
+        // set delete mode
+        if (mode != null) {
+            config.getDeleteConfig().setMode(mode == Mode.terminate ?
+                    DeleteMode.TERMINATE :
+                    DeleteMode.DELETE);
+        }
+
+        try {
+            Deleter deleter = new Deleter();
+            if (deleteListOption != null) {
+                deleter.doDelete(deleteListOption.toDeleteListParser());
+            } else {
+                // set user-defined query options
+                if (queryOption != null) {
+                    config.getDeleteConfig().setQuery(queryOption.toQueryConfig());
+                }
+
+                deleter.doDelete();
+            }
+
+            log.info("Database delete successfully finished.");
+        } catch (DeleteException e) {
+            log.error(e.getMessage(), e.getCause());
+            log.warn("Database delete aborted.");
+            return 1;
+        } finally {
+            database.disconnect(true);
+        }
+
+        return 0;
+    }
+
+    @Override
+    public void preprocess(CommandLine commandLine) {
+        if (queryOption != null && deleteListOption != null) {
+            throw new CommandLine.ParameterException(commandLine,
+                    "Error: Query options and delete file options are mutually exclusive (specify only one)");
```

Comment:
The querying settings in the config file and the delete file options are also mutually exclusive. When the option `--config` is specified, maybe the CLI should prints some warning/hints.

Question: why the query options and the delete file are mutually exclusive, cannot be combined?

Reply:
1. Yes, technically we could combine a delete list with query options. But I think the typical use case is to use either a delete list or a query expression. I would prefer to keep them separate and mutually exclusive. 

2. The CLI options always take precedence over corresponding settings in a provided config file. This is true for all options (not just query options) and all commands. This is a correct and intended behaviour, so I don't think we should record warnings in the log. But I agree that this must be explained, and I think the documentation is the right place to do so.

Reply:
1. There is an important use case of using delete list against _versioned_ database. In this case, the combination of  `featureVersionFilter` and delete list would be required. 

2. good

Reply:
But in your example, wouldn't an additional `--exclude-terminated` flag be sufficient?

Reply:
`--exclude-terminated` might not be sufficient enough for the case that a user want to physically delete some very old feature versions based on certain timestamps?

Reply:
Hm, but if your use case is really to identify the features to be deleted in such a precise way, then you can always go with an XML-based query expression or even directly use a SQL filter. For me, the use case for a delete list is to delete/terminate the current feature version only, not old versions. 

But yes, this understanding is in fact debatable. Maybe we should ask for input from others?

Reply:
yes, some inputs from others are helpful, especially from the view of users, not developers :-)
I think we could leave it for now. Your currently implmentation already covers most use cases. We can change the behaviour in future when we really need such precise delete in practical use cases.